src/components/Faucet/index.tsx

```diff
@@ -99,59 +107,91 @@ export default function FaucetPage({
     signMessageSuccess,
     signMessageData,
     signMessageError,
-    signMessageLoading
+    signMessageLoading,
+    faucetTokenRequest
   ])
 
   useEffect(() => {
-    const fetchAddress = async () => {
+    const fetchAddressAndNetwork = async () => {
       try {
         const provider = new ethers.providers.Web3Provider(window.ethereum)
         const signer = provider.getSigner()
         const address = await signer.getAddress()
-        setDid(address)
+        const network = await provider.getNetwork()
+        setAddress(address)
+        setNetwork(networkNameMap[network.chainId] || 'Unknown')
       } catch (error) {
         LoggerInstance.error(error)
       }
     }
 
-    fetchAddress()
+    fetchAddressAndNetwork()
 
-    if (didQueryString) {
-      setDid(didQueryString)
-      handleVerify()
+    if (window.ethereum) {
+      window.ethereum.on('accountsChanged', fetchAddressAndNetwork)
+      window.ethereum.on('chainChanged', fetchAddressAndNetwork)
+    }
+
+    return () => {
+      if (window.ethereum) {
+        window.ethereum.removeListener(
+          'accountsChanged',
+          fetchAddressAndNetwork
+        )
+        window.ethereum.removeListener('chainChanged', fetchAddressAndNetwork)
+      }
```

Comment:
suggestion: we should rather make use of wagmi react hooks here

see [useAccount](https://wagmi.sh/react/api/hooks/useAccount) / [useNetwork](https://1.x.wagmi.sh/react/hooks/useNetwork)

Reply:
This will be addressed in a subsequent pull request.